### PR TITLE
Cleanup 2 deprecation warnings

### PR DESCRIPTION
**Pull request by kanunnikau**
RKHTTPRequestOperation: remove unused (and possibly dangerous) NSURLConnectionDelegate method. If it happens to be used it will try to trigger this method on a super - but no superclass implements it, so it will crash.
RKMappingOperation: remove RestKit 'mappingOperationShouldSkipPropertyMapping:' deprecation warning by replacing with two corresponding methods (attribute, relationship mapping). Is it a suitable fix?

### CODE REVIEW

File: Code/ObjectMapping/RKMappingOperation.m [hunk 1]
```diff
@@ -1222,10 +1222,9 @@ - (void)main
         }
     }
     
-    BOOL canSkipProperties = [dataSource respondsToSelector:@selector(mappingOperationShouldSkipPropertyMapping:)] && [dataSource mappingOperationShouldSkipPropertyMapping:self];
-    BOOL canSkipAttributes = canSkipProperties;
-    BOOL canSkipRelationships = canSkipProperties;
-    if ([dataSource respondsToSelector:@selector(mappingOperationShouldSkipRelationshipMapping:)]) {
+  BOOL canSkipAttributes = [dataSource respondsToSelector:@selector(mappingOperationShouldSkipAttributeMapping:)] && [dataSource mappingOperationShouldSkipAttributeMapping:self];
```
Comment (segiddins) [inline Code/ObjectMapping/RKMappingOperation.m]: Please use consistent whitespace

File: Code/Network/RKHTTPRequestOperation.m [hunk 2]
```diff
@@ -112,13 +112,6 @@ - (NSError *)error
 
 #pragma mark - NSURLConnectionDelegate methods
 
-- (void)connection:(NSURLConnection *)connection didReceiveAuthenticationChallenge:(NSURLAuthenticationChallenge *)challenge
```
Comment (segiddins) [inline Code/Network/RKHTTPRequestOperation.m]: why was this removed ? Remember, RK supports back to iOS 5.1.1, which may well still need this method

Comment (kanunnikau) [inline Code/Network/RKHTTPRequestOperation.m]: It calls same method on super there. I did not see any super class implementing it, so i thought that it would crash in that case.
